fs: Implement fstab devicetree integration for EXT2

bas-archembedded opened the pull request:
Device tree and/or automount support was not present and might be useful when for example the settings module needs early access to a filesystem. This patch adds support to mount an EXT2 filesystem automatically. The implementation chosen is to provide the drive name matching a disk drive specified in the device tree ie on an (e)MMC or SD card.

bas-archembedded committed:
fs: Implement fstab devicetree integration for EXT2

Device tree and/or automount support was not present and might be useful
when for example the settings module needs early access to a filesystem.
This patch adds support to mount an EXT2 filesystem automatically. The
implementation chosen is to provide the drive name matching a disk
drive specified in the device tree ie on an (e)MMC or SD card.

Signed-off-by: Bas van Loon <bas@arch-embedded.com>

rruuaanng commented on dts/bindings/fs/zephyr,fstab,ext2.yaml:
```diff
@@ -0,0 +1,16 @@
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+    Description of pre-defined EXT2 file systems.
```
```suggestion
    pre-defined EXT2 file systems.
```

bas-archembedded replied:
Updated. :)

de-nordic reviewed (changes requested):
Can we have fs sample for this, the same way FAT has?

bas-archembedded committed:
samples: fs: Add EXT2 automount sample.

Add a simple sample that tests the functionality of the EXT2 automount
feature. The sample creates a ramdisk and writes some content and tries
to read it back after unmount and mount operation.

Signed-off-by: Bas van Loon <bas@arch-embedded.com>

de-nordic commented on samples/subsys/fs/ext2_fstab/README.rst:
```diff
@@ -0,0 +1,36 @@
+Overview
+***********
+
+This sample shows how to define a ext2 fstab entry in the devicetree.
```
```suggestion
This sample shows how to define an ext2 fstab entry in the devicetree.
```

de-nordic commented on samples/subsys/fs/ext2_fstab/README.rst:
```diff
@@ -0,0 +1,36 @@
+***********
+
+This sample shows how to define a ext2 fstab entry in the devicetree.
+This scenario uses a ext2 on a RAM disk. The sample is run on the
+``native_sim`` board.
```
```suggestion
The sample is run on the``native_sim`` board and RAM disk is used for ext2 storage.
```

bas-archembedded replied:
Updated.
